File: src/main/java/com/zerozero/auth/presentation/response/LoginResponse.java
package com.zerozero.auth.presentation.response;

import com.zerozero.user.domain.response.UserResponse;

public record LoginResponse(
        UserResponse user,

        TokenResponse token
) {
    public static LoginResponse of(UserResponse user, TokenResponse token) {
        return new LoginResponse(user, token);
    }
}

File: src/main/java/com/zerozero/auth/presentation/response/LoginResponse.kt
package com.zerozero.auth.presentation.response

import com.zerozero.user.domain.response.UserResponse

data class LoginResponse(
    val user: UserResponse,
    val token: TokenResponse
) {
    companion object {
        fun of(user: UserResponse, token: TokenResponse) = LoginResponse(user, token)
    }
}
